### ISSUE TITLE
name ID 4.3.1 (Windows FullName) in OTFs — current recommendations?

### ISSUE THREAD

**Issue by twardoch** [adobe-type-tools/afdko#1066]
In FontLab Studio 5, we had the option "Use PostScript FontName as FullName on Windows" which, only in OTFs, made "name" ID 4.3.1 (FullName) same as 6.3.1 (PostScript FontName). 

I know it was needed for something back around 2005. What’s the current practice? Does MakeOTF still do this? Does Adobe still do this? Or, if it’s no longer needed, when did it cease to be needed? And why was it needed in the first place? 

The discussion in the thread that I wrote years ago seems inconclusive :/ 

http://forum.fontlab.com/fontlab-studio-tips-and-tricks/font-family-naming-in-fontlab-studio-5/

**Comment by readroberts** [comment]
makeotf will still do this if you use the old style names, but it is no longer necessary (hasn't been in since at the latest Windows 7) , and it is no longer the makeotf default to make Windows namedID 4 == named ID 6.

**Comment by twardoch** [comment]
Thanks!

**Comment by twardoch** [comment]
To make the record complete:

> From: Read Roberts <rroberts@adobe.com>
> Sent: Thursday, November 13, 2003 9:20 AM
> To: opentype@topica.com
> Subject: Re: [OpenType] Full Naming Issues
>
> The "someone" is incorrect.  The reason for this special case is old OS support issues.
> Under Windows 95/98,  applications could not find out the PS name of an
> OpenType/CFF font  (i.e. the contents for name table name ID 6). Under several
> circumstances, this information was required.  The Adobe developers of the
> Type1/OpenType-CFF font driver chose the name table Windows platform Full Name
> name id  as an additional  place to put the PS name; programs can get the name of
> this field.  It was a bad choice, but it was the least bad of the options.
>
> A side effect of this is that the PS name of the OTF shows up in the Preview window
> rather than the font Full Name, however,  I would describe this  as a  (minor) bug in
> the Preview window,  rather than a feature.
>
> Perhaps someone else can answer if keeping this practice is still a requirement. I am
> reasonably sure, but not certain,  that current Adobe applications under Windows
> 2000 do not require this anymore.
>
>


> From: Read Roberts <rroberts@adobe.com>
> Sent: Thursday, November 13, 2003 10:11 PM
> To: opentype@topica.com
> Subject: Re: [OpenType] Full Naming Issues
>
> Unfortunately, I have since learned that the issue is with the API for the Win32 GDI,
> not with API's specific  to Win 95/98/ME.
> Dependency on the  Win32 GDI is not going away anytime soon.
>
> Any program which needs to enumerate fonts by their PostScript name can get to the
> PS name of an OpenType/CFF only if it has been placed in the Windows name table
> Full Name name id.  (the PS name then shows up in "otmpFaceName" in
> OUTLINETEXTMETRIC and "elfFullName" in
> ENUMLOGFONT.)
>
>   Following the current spec in this regard seems likely  to be necessary for the
> forseeable future.
>
> My thanks to David Arnold, who supplied me with more detail.
>
>

**Comment by readroberts** [comment]
David's comments about the API are certainly still correct. However, I think we would know by now if  any clients still depended on having the PS name in Windows name ID 4. In October 2010, we discussed this issue with MS and Apple, and the requirement to set Windows name ID 4 to the PS name for CFF fonts was removed in an update to OpenType spec 1.6, and we rebuilt the Adobe Type Library with Windows Name IF 4 set to the Full Name value, same as for TTF fonts. This was driven partly because MS and Apple had already been making their fonts without the special case for Windows Name ID 4 in OTF fonts, and because we all wanted to remove special cases for OTF vs CFF. This special practice has not been a requirement for nearly a decade.
